web/src/routes/api/account/gmail/remove-email-labels/+server.ts

```diff
@@ -0,0 +1,198 @@
+  return updateSuccessful;
+};
+
+/**
+ * Checks if a given thread ID corresponds to a Gmail thread by attempting to retrieve the thread details using the Gmail API.
+ *
+ * @async
+ * @param {string} accessToken - The access token for the authenticated user.
+ * @param {string} threadId - The ID of the thread to check.
+ * @returns {Promise<boolean>} - A promise that resolves with a boolean indicating whether or not the thread ID corresponds to a Gmail thread.
+ * @throws {Error} - Throws an error if the API request fails or if the response is invalid.
+ */
+const isGmailThreadIdValid = async (accessToken: string, threadId: string): Promise<boolean> => {
+  // The Gmail thread endpoint, which gives the details of a specific Gmail thread
+  const threadEndpoint = `https://gmail.googleapis.com/gmail/v1/users/me/threads/${threadId}`;
+
+  try {
+    // Query the Gmail thread endpoint with the specified thread ID
+    const response = await fetch(threadEndpoint, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        Accept: 'application/json',
+      },
+    });
+
+    // If the API returns a valid response, the thread ID is valid and corresponds to a Gmail thread
+    return response.ok;
+    
+  } catch (err) {
+    // If the API request fails or the response is invalid, throw an error
+    throw new Error(`Failed to retrieve information for thread ID: ${threadId}, corresponds to a Gmail thread: ${err.message}`);
+  }
+};
+
+/**
+ * Main handler function for this POST request route
+ *
+ * @param request - The incoming request object
+ * @param locals.getSession - function used to retrieve the current user's session.
+ * @param locals.supabase - the supabase client
+ * @returns {Response} - A response indicating whether or not the operation was successful.
+ * @throws Throws an error if the user is not authorized or if an unexpected error occurs.
+ */
+export const POST: RequestHandler = async ({ request, locals: { getSession, supabase } }) => {
+  
+  // Get the current user's session
+  const session = await getSession();
+  if (!session) throw error(401, 'unauthorized');
+
+  // retrieve the necessary information from the request 
+  const { email, email_thread_id: threadId } = await request.json();
+
+  if (!email || !threadId) {
+    throw error(400, 'Could not find both email and email_thread_id from request json');
+  }
+
+  // Get a refreshed Google access token for the current user
+  let accessToken = '';
+  try {
+    accessToken = await getRefreshedGoogleAccessToken(supabase, email);
+  } catch (err) {
+    throw error(500, err instanceof Error ? err : 'Could not retrieve access token');
+  }
+
+  // check the thread actually exists in Gmail before attempting the update
+  const isValidGmailThread = await isGmailThreadIdValid(accessToken, threadId)
+  if (!isValidGmailThread) {
+    return new Response({status: 404, statusText: "No corresponding Gmail thread"});
```

Comment:
Use `error` for 4xx and 5xx response code
```suggestion
    throw error(404, 'No corresponding Gmail thread');
```